File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length < 2) return 0;
        int maxProfit = 0;
        int minPrice = 0;
        int i = 0;
        int j = 1;
        while(i < prices.length -1){
            if(prices[i] > prices[i+1]){
                i++;
                j = i+1;
                continue;
            }


            if(maxProfit < prices[j] - prices[i]){
                maxProfit = prices[j] - prices[i];
                minPrice = prices[i];
            }
            j++;
            if(j == prices.length){
                i++;
                j = i + 1; 
            }
        }


        return maxProfit;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        if(prices.size < 2) return 0
        var totalProfit = 0
        var i = 0
        var j = 1
        var curProfit = if(prices[i] < prices[j]) prices[j] - prices[i] else 0  
        while(j < prices.size){
            if(prices[j] < prices[j - 1]){
                totalProfit += curProfit
                i = j++
                curProfit = 0
                continue
            }
            if(prices[i] < prices[j]){
                curProfit = max(curProfit, prices[j] - prices[i]) 
            }else{
               totalProfit += curProfit
               curProfit = 0
               i = j 
            }
            j++

        }

        return totalProfit + curProfit
    }
}
